src/pydata_sphinx_theme/toctree.py

```diff
@@ -289,22 +325,22 @@ def add_header_level_recursive(ul, level):
             if ul is None:
                 return
             if level <= (context["theme_show_toc_level"] + 1):
-                ul["class"] = ul.get("class", []) + ["visible"]
+                ul["class"] = [*ul.get("class", []), "visible"]
             for li in ul("li", recursive=False):
-                li["class"] = li.get("class", []) + [f"toc-h{level}"]
+                li["class"] = [*li.get("class", []), f"toc-h{level}"]
```

Comment:
these (and similar changes below) were suggested by Ruff in my IDE, and TBH I did it mostly to make the squiggly underlines go away, they were distracting me. I guess there's probably a very small performance benefit?